Source/UI/CommonUI/Protocols/Elements/ElementBorderStyle.swift

```diff
@@ -0,0 +1,11 @@
+import UIKit
+
+public protocol ElementBorderStyle {
+    var cornerRadius: CGFloat { get }
+    var borderWidth: CGFloat { get }
+    var normalColor: UIColor { get }
+    var focusColor: UIColor { get }
+    var errorColor: UIColor { get }
+    var edges: UIRectEdge? { get }
+    var corners: UIRectCorner? { get }
```

Comment:
I don't think we need to have optional edges and corners? They should always be provided, and being option sets, even values as [] will work.